### PR TITLE
WL-3959: Citation edit and remove links not working when dropped in nest

### PR DESCRIPTION
**Pull request by nicholaswilson100**
The fix for this is to fix the edit and remove link parameters in _nestableCitation.vm.
Before for example collectionId was null, now we are passing in citationCollectionId and the location of the citation

Also, the View Screen was showing nested citations (we only to show unnested citations)

Also I've renamed various variables/methods to more accurately describe what they do/are

### CODE REVIEW

File: citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java [hunk 1]
```diff
@@ -294,7 +294,7 @@ protected void handleViewRequest(HttpServletRequest req, HttpServletResponse res
     		String description = properties.getProperty( ResourceProperties.PROP_DESCRIPTION );
     		
      		String citationCollectionId = new String( resource.getContent() );
-    		CitationCollection collection = CitationService.getCollection(citationCollectionId);
+    		CitationCollection collection = CitationService.getInstance().getUnnestedCitationCollection(citationCollectionId);
```
Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: Good to get rid of covers but it's generally better avoid the `getInstance()` call by using `ComponentManager.get(Class)`

File: citations-api/api/src/java/org/sakaiproject/citation/api/CitationService.java [hunk 2]
```diff
@@ -229,15 +229,15 @@
 	 */
 	public CitationCollectionOrder getNestedCollection(String citationCollectionId);
 	/**
-	 ** Removes the section of a citation collection and its children
+	 ** Removes the section of a citation collection and its children or removes a citation
 	 * @param collectionId
 	 * @param locationId
 	 */
-	public void removeSection(String collectionId, int locationId);
+	public void removeSectionOrCitation(String collectionId, int locationId);
```
Comment (buckett) [inline citations-api/api/src/java/org/sakaiproject/citation/api/CitationService.java]: `removeLocation()` as a name as it doesn't really care what's at the location does it?

Comment (buckett) [inline citations-api/api/src/java/org/sakaiproject/citation/api/CitationService.java]: Is this designed to remove items from the non-nested sections as well?

Comment (nicholaswilson100) [inline citations-api/api/src/java/org/sakaiproject/citation/api/CitationService.java]: No just nested citations.  Both a nested citation and an unnested citation have a 'remove' link which goes to CitationHelperAction.doRemoveSelectedCitations(RunData) where the locationId is checked for; if it has a locationId (i.e., it's nested), it deletes the citation from the nested collection ; if it's null or 0, it deletes it from the usual unnested collection using the usual flow.

**Comment by nicholaswilson100** [comment]
I've made some changes and rebased.

**Pull request merged**